cloud/vsphere/provisioner/govmomi/exists.go

```diff
@@ -3,13 +3,19 @@ package govmomi
 import (
 	"context"
 
+	"github.com/golang/glog"
 	"github.com/vmware/govmomi/vim25/mo"
 	"github.com/vmware/govmomi/vim25/types"
 	vsphereutils "sigs.k8s.io/cluster-api-provider-vsphere/cloud/vsphere/utils"
 	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 )
 
 func (vc *Provisioner) Exists(cluster *clusterv1.Cluster, machine *clusterv1.Machine) (bool, error) {
+	glog.Infof("govmomi.Actuator.Exists %s", machine.Spec.Name)
+	if machine.Status.NodeRef != nil {
```

Comment:
I know you added this to resolve the issue with `clusterctl` CLI getting hung.. 
But IMHO in our provider's `Exists` method we should be truly checking the existence of the resource on the provider infrastructure and should be reporting on that. The `NodeRef` property is neither added nor controlled by the actuator, thus the actuator exists should not be using this check. 
As for the `clusterctl` CLI issue, I am still not sure how this check is helping, since the `NodeRef` does not get set until the pivot has happened, and the actuator's `Exists` method must have to return `true` for the master node before we could have ever triggered the pivot operation.
I think we should be fixing the code making up the `clusterctl` CLI that has logic making certain improper assumptions to begin with.